angr/knowledge_plugins/functions/function.py

```diff
@@ -1487,6 +1487,18 @@ def apply_definition(self, definition, calling_convention=None):
         else:
             raise TypeError("calling_convention has to be one of: [SimCC, type(SimCC), None]")
 
+    def functions_called(self) -> Set['Function']:
+        """
+        :return: The set of all functions that can be reached from the function represented by self.
+        """
+        def _called(function_address) -> Set[int]:
+            successors = set(self._function_manager.callgraph.successors(function_address))
+            called_functions = successors
+            for a in [ _called(s) for s in successors]:
```

Comment:
I think a better way is the following:

```
for s in successors:
    called_functions |= _called(a)
```

You avoid creating a list this way.